cfpb/consumerfinance.gov
# Paying for College disclosures: Replace sticky-kit with customized ESM fork

sticky-kit is a 6-year old package that adds a jquery plugin to create a `stick_in_parent` function for making parts of the paying for college disclosure app stick in view while scrolling. With the move to ESM https://github.com/cfpb/consumerfinance.gov/pull/7412, this old dependency couldn't be required after jquery and was changed to an import. However, the `import(…)` is asynchronous and it's not loading in time/at all to be available, leading to this error on the page:

<img width="438" alt="Screen Shot 2023-01-06 at 8 21 10 AM" src="https://user-images.githubusercontent.com/704760/211021185-d8242bc9-a627-4979-943b-2439ba97e5f2.png">

Since this is a stale project, and relatively small (one function), I have opted to copy the plugin directly into our codebase, modifying it to be an ESM module, and importing it that way. Versus continuing to use the sticky-kit npm module.

## Additions

- ESM version of sticky-kit.


## Removals

- Old sticky-kit npm module.

## How to test this PR

1. Navigate to [Dynamic Disclosures locally](http://localhost:8000/paying-for-college2/understanding-your-financial-aid-offer/offer/?iped=154022&pid=BAHCA&totl=70950&tuit=14160&hous=7109&book=1150&tran=500&othr=4743&pelg=0&schg=&stag=&othg=&mta=&gib=&wkst=0&ppl=&parl=&perl=&subl=3500&unsl=6000&gpl=&prvl=&prvi=&prvf=&insl=&insi=&inst=&oid=4D0B9265B45716DF4B81EEB4BE117E1DF69871AF) and verify that there is no dev console error when the page loads.
